### ISSUE TITLE
MongooseIM Suse Linux Enterprise server Intalltion error

### ISSUE THREAD

**Issue by vickydhas** [esl/MongooseIM#1137]
MongooseIM version: 2.0.0
Installed from: github
Erlang/OTP version: 18

I was able to build the Mongoose server on Suse Linux after all dependancies and library support , the build is also complete with no error but the mongooseim folder was not generated.

```
developer@suse-linux-enterprise:~/mim/MongooseIM> sudo make
./rebar get-deps > deps.log 2>&1 || (cat deps.log; exit 1)
./rebar  compile > compile.log 2>&1 || (cat compile.log; exit 1)

developer@suse-linux-enterprise:~/mim/MongooseIM> ls
apps             COPYING                   doc           Makefile     rebar                test
CHANGELOG.md     deps                      ebin          mkdocs.yml   rebar.config         tools
compile.log      deps.log                  elvis.config  priv         rebar.config.script  VERSION
configure.out    dialyzer.ignore-warnings  examples      readlink.sh  rel
CONTRIBUTING.md  dialyzer.mk               index.html    README.md    src

developer@suse-linux-enterprise:~/mim/MongooseIM> cd rel/

developer@suse-linux-enterprise:~/mim/MongooseIM/rel> ls

configure.vars.config  files  reltool.config.script  reltool_vars  vars.config.in

```

**Comment by vickydhas** [comment]
To my understanding its requireing libexpat1-dev package , GCC i installed to move ahead , but suse is having support to libexpat1 not libexpat1-dev how to get this.

**Comment by kklocek** [comment]
Hi @vickydhas !

You need to run commands `make` and after that `make rel` to generate MongooseIM build.

**Comment by vickydhas** [comment]
Yes my apology i forgot this , but i get this build rel failure if i run $sudo make rel / $make rel command after success $make command.

```
==> rel (generate)
ERROR: Reltool support requires the reltool application to be installed!ERROR: generate failed while processing /home/vickydhas/mim/MongooseIM/rel: rebar_abort
Makefile:101: recipe for target 'rel' failed
make: *** [rel] Error 1

```

**Comment by vickydhas** [comment]
[ejabberd.txt](https://github.com/esl/MongooseIM/files/677346/ejabberd.txt)
[erlang.log.1.txt](https://github.com/esl/MongooseIM/files/677347/erlang.log.1.txt)
[run_erl.txt](https://github.com/esl/MongooseIM/files/677348/run_erl.txt)

**Comment by kklocek** [comment]
I am not a developer here but in logs I saw:
`{"Kernel pid terminated",application_controller,"{application_start_failure,fast_tls,{{shutdown,{failed_to_start_child,fast_tls,{open_error,-10}}},{fast_tls_app,start,[normal,[]]}}}"}`

Maybe some silly questions:
1. Ensure you have installed openssl
2. Do `make clean`
3. Again `make` and `make rel`

**Comment by fenek** [comment]
Hi @vickydhas 

Please check if OpenSSL development package is installed as well. I don't know the exact name for SuSE but most probably it will be `openssl-devel` or `openssl-dev`

**Comment by vickydhas** [comment]
Hello Piotr
​
​I have installed this , i am able to make $sudo make and $sudo make rel ,
command happens with no error , but the rel folder doesn't get created.​

**Comment by vickydhas** [comment]
Thanks for this valuable input I am able to install and register the user using mnesia, let me check with the MS SQL as its not integrating with this chat server even after following guidelines of Backend Database configuration on MongooseIM  Read Docs.

**Comment by vickydhas** [comment]
Closing this bug as i have resolved it by adjusting the odbc.ini, idbcinst.ini and freetds.conf settings. 

I have to enable the ports in local machine for the outside users to access in linux. I used Yast , Security & Users -- Firewall and Advanced settings changed to port 5222 enabled and 1433.

Hope this help some one. 

Now facing this issue where offline message are stored in DB and seen in DB when queried and emptied when user comes online. but the actual message is not received / sent from MIM to user. also checked the ejabberd.log. 

this settings is confirmed to add {backend, odbc} to mod_offline module. FYI.
